### ISSUE TITLE
Write Go Coding Style Doc

### ISSUE THREAD

**Issue by rob05c** [apache/incubator-trafficcontrol#2077]
Most of the Go Coding Style is covered in https://golang.org/doc/effective_go.html and https://github.com/golang/go/wiki/CodeReviewComments but not everything.

For example, two notable things:

* don't use relative import paths. They are pain.
* don't vendor `golang.org/x`, treat it as part of the compiler, because old `x` breaks with new compilers.

Another example is, that the Go community has widely agreed that named return params are confusing and shouldn't be used, even though the Go authors like them.

**Comment by dneuman64** [comment]
@ocket8888 is this stuff covered in the API 3.0 doc?  If so we can close this.

**Comment by ocket8888** [comment]
No, this is a request for a style guide for writing all Go code in ATC. The API v3 doc is only about the HTTP API - it doesn't touch Go coding style.
